refactor!: remove experimental features from release builds

gustavovalverde opened the pull request:
## Motivation

Experimental features are no longer built into release images. Users must now build custom images with specific features if needed.

Closes https://github.com/ZcashFoundation/zebra/issues/9210
Closes https://github.com/ZcashFoundation/zebra/issues/9204
Partially https://github.com/ZcashFoundation/zebra/issues/7415
Depends-On: #8923

## Solution


- Remove experimental build job from release workflow
- Remove experimental features from CI matrix
- Update documentation to reflect new feature strategy
- Simplify Dockerfile by removing experimental feature args
- Keep RUN_ALL_EXPERIMENTAL_TESTS for testing purposes only

### Notes

Migration: Users who need specific features should:
1. Build custom images using --build-arg FEATURES="feature1 feature2"
2. Use the documented feature flags in their builds
3. Refer to the new documentation for building with custom features

### PR Author's Checklist

<!-- If you are the author of the PR, check the boxes below before making the PR
ready for review. -->

- [X] The PR name will make sense to users.
- [X] The PR provides a CHANGELOG summary.
- [X] The solution is tested.
- [X] The documentation is up to date.
- [X] The PR has a priority label.

### PR Reviewer's Checklist

<!-- If you are a reviewer of the PR, check the boxes below before approving it. -->

- [ ] The PR Author's checklist is complete.
- [ ] The PR resolves the issue.

gustavovalverde committed:
refactor!: remove experimental features from release builds

BREAKING CHANGE: Experimental features are no longer built into release images.
Users must now build custom images with specific features if needed.

- Remove experimental build job from release workflow
- Remove experimental features from CI matrix
- Update documentation to reflect new feature strategy
- Simplify Dockerfile by removing experimental feature args
- Keep RUN_ALL_EXPERIMENTAL_TESTS for testing purposes only

Migration: Users who need specific features should:
1. Build custom images using --build-arg FEATURES="feature1 feature2"
2. Use the documented feature flags in their builds
3. Refer to the new documentation for building with custom features

Closes  #9210
Closes #9204
Partially #7415

gustavovalverde committed:
revert(ci): Add support for experimental features in test runs

Enable passing experimental features to Docker test runs by:
- Updating workflow configuration to pass EXPERIMENTAL_FEATURES
- Modifying entrypoint script to include experimental features during test execution
- Documenting new experimental test category in workflow README

upbqdn reviewed (commented):
I think it'll be easier if we merge #8923 first since it also simplifies passing Rust features to the images using only a single variable.

upbqdn commented:
It would be good to get this in before the next release.

gustavovalverde committed:
Merge branch 'main' into remove-experimental-build

gustavovalverde committed:
refactor(docs): streamline Docker documentation and feature testing strategy

- Reorganized and clarified advanced usage instructions for building Docker images with custom features, including metrics support.
- Enhanced the presentation of alternatives and configuration details for Docker usage.

gustavovalverde commented:
This is ready now

upbqdn commented on CHANGELOG.md:
```diff
@@ -5,6 +5,13 @@ All notable changes to Zebra are documented in this file.
+
+- Remove experimental features from release builds and workflows
+
+
```
This isn't a breaking change because the last experimental build we published was 1.8, which was 9 months ago.

```suggestion
```

upbqdn commented on book/src/user/docker.md:
```diff
@@ -54,10 +52,34 @@ docker build \
 
+### Alternatives
+
+See [Building Zebra](https://github.com/ZcashFoundation/zebra#building-zebra) for more information.
```
Invalid link

```suggestion
See [Building Zebra](https://github.com/ZcashFoundation/zebra#manual-build) for more information.
```

upbqdn commented on book/src/user/docker.md:
```diff
@@ -54,10 +52,34 @@ docker build \
+
+## Advanced usage
+
+You're able to specify various parameters when building or launching the Docker image, which are meant to be used by developers and CI pipelines. For example, specifying the Network where Zebra will run (Mainnet, Testnet, etc), or enabling features like metrics with Prometheus.
```
I'd say selecting between Mainnet and Testnet and monitoring is basic functionality for all users. I'd rather avoid orienting it at advanced users or devs.

```suggestion
```

upbqdn commented on book/src/user/docker.md:
```diff
@@ -54,10 +52,34 @@ docker build \
+
+Zebra supports various features that can be enabled during build time using the `FEATURES` build argument:
+
+For example, if we'd like to enable metrics on the image, we'd build it using the following `build-arg`:
```
This will become outdated once we enable the `prometheus` by default.

gustavovalverde replied:
I can't think of another full example right now. If this is a blocker, I can prepare an example later. If not, we can fix it with the PR that enables prometheus by default, as a lot of documentation and files (including the docker-compose) will require update with that change.

gustavovalverde committed:
Apply suggestions from code review

Co-authored-by: Marek <mail@marek.onl>

upbqdn commented on book/src/user/docker.md:
````diff
@@ -17,8 +17,6 @@ If you want to preserve Zebra's state, you can create a Docker volume:
 docker volume create zebrad-cache
 ```
 
-And mount it before you start the container:
````
Why this change?

gustavovalverde replied:
Reverted [chore: revert deleted sentence](https://github.com/ZcashFoundation/zebra/pull/9222/commits/f737f7053cf601a191dccea08ce6334b2620a629)

upbqdn commented on book/src/user/docker.md:
```diff
@@ -54,10 +52,31 @@ docker build \
 <https://docs.rs/zebrad/latest/zebrad/index.html#zebra-feature-flags>.
 
-## Configuring Zebra
+## Configuring Zebra with Docker Compose
```
This section covers general config options that don't pertain only to Docker Compose.

```suggestion
## Configuring Zebra
```

gustavovalverde committed:
chore: revert deleted sentence

gustavovalverde committed:
Update book/src/user/docker.md

Co-authored-by: Marek <mail@marek.onl>
